astropy/io/fits/header.py

```diff
@@ -1460,6 +1467,9 @@ def insert(self, key, card, useblanks=True, after=False):
                 f"(keyword, value, [comment]) tuple; got: {card!r}"
             )
 
+        if self._validate_card(card) is None:
```

Comment:
I think in the new scheme this should become
```
card = self._validate_card(card)
if card is None:
    return
```
Otherwise any changes made would not actually be used!  (same above)

Reply:
Or is `_validate_card` allowed to make in-place changes to the `card`? That would definitely make sense too.

Reply:
I think it should be allowed to return a new card so have updated following your suggestion